site/zenodo_rdm/legacy/serializers/schemas.py

```diff
@@ -63,6 +68,23 @@ def dump_resource_type(self, result, original, **kwargs):
                 result[f"{upload_type}_type"] = resource_type_id.split("-")[-1]
         return result
 
+    def dump_license(self, data):
+        """Dumps license field."""
+        license = data.get("rights", [])
+
+        if not license:
+            return missing
+
+        # Zenodo legacy only accepts one right.
+        license = license[0]
+
+        legacy_license = rdm_to_legacy(license["id"])
+
+        # TODO if does not map, what to do?
+        # TODO for deserialization we use a custom one.
```

Comment:
I think the format is just to return `{... license: id ...}`, unless you mean something else?